samples/subsys/tracing/sample.yaml

```diff
@@ -37,3 +37,8 @@ tests:
   sample.tracing.transport.native_posix.ctf:
     platform_allow: native_posix
     extra_args: CONF_FILE="prj_native_posix_ctf.conf"
+  sample.tracing.percepio:
+    platform_allow: frdm_k64f
```

Comment:
I'm wondering if we have any reasons for such testing scope limitations? With surrent configuration it looks like we can build it on any platform.

Reply:
this is just a build configuration, not a test in any way. Building it on multiple platforms is just going to be wasteful and time consuming without any value.

Reply:
does this sample run on nsim? I did not know it does, this is not only supports transports that are ARM specific. This is just testing integration of the module, it is neither a toolchain test, nor a functional test. Adding nism here does not make sense at all.

Reply:
> this sample only supports transports that are ARM specific

Does it? TraceRecorder library supports multiple streamports, and as I can see now the is no streamport hardcoded, so for each target the preferred one is selected.

For the `frdm_k64f` it's `PERCEPIO_TRC_CFG_STREAM_PORT_RTT`.
for the `nsim_hs` it is `PERCEPIO_TRC_CFG_STREAM_PORT_RINGBUFFER`.

So it's ok to build it for nsim, moreover the meaningful option for nsim is selected.

Reply:
let me ask again, can you make this sample work with nsim and percepio?

Reply:
> can you make this sample work with nsim and percepio?

Yes. I've just tried it and it works. I can show screenshots with this sample trace visualization if you are still in doubt :)

Reply:
> Yes. I've just tried it and it works. I can show screenshots with this sample trace visualization in percepio if you are still in doubt :)
> 
> So I'd propose to add `nsim_hs` to platform_allow.

I am not in doubt, and it is nice to see that it works. I however still disagree that we need to add any other platforms to a sample, this is just a check that we can be build with, not a test that needs to be expanded to each supported hardware. Thats why all the other sections in this file are limited to just the minimum.

Reply:
> this is just a check that we can be build with

Indeed, and I prefer to have a check that we can build it with mwdt toolchain. Having the only frdm_k64f platfom doesn't allow us to achieve it. There is a lot of situations where something which can be build with GCC can't be build / produce warnings with other toolchain.

Moreover in this sample we have console output check - so it's not only the build test. We are checking that we can build sample with tracing enabled and TraceRecorder library as a backend using mwdt toolchain, and we checks that sample is still working with that.

If it’s not tested, it doesn’t work.

Reply:
should this be`integration_platforms`?

Reply:
If we add `integration_platforms: frdm_k64f` instead of `platform_allow: frdm_k64f` this would be 100% fine for me.

Reply:
integration_platforms would mean that you will have to go in and filter everything that is not supported, and there are many platforms that are not supported and fail to build. This is a sample and I do not intend to spend more time on it than it is needed. If you want to add anything here, it needs to be verified that it works, i.e. you are welcome to come in after this is merged, add the platform you wish to add, and *document* in the README.rst how to use it with any of the tracing methods supported. Just adding more build targets here does not make sense.

Reply:
>  If you want to add anything here, it needs to be verified that it works, i.e. you are welcome to come in after this is merged, add the platform you wish to add, and document in the README.rst how to use it with any of the tracing methods supported. 

I'm fully OK to add it separately by myself (in this or separate testcase). I'm just afraid that it'll also just end with similar discussion.

Reply:
We could filter on `CONFIG_HAS_SEGGER_RTT`

Reply:
> We could filter on CONFIG_HAS_SEGGER_RTT

Well, generally it doesn't require SEGGER_RTT as well as any other HW (in case of ringbuffer streamport). So filter on `CONFIG_HAS_SEGGER_RTT` won't we really helpful :)

Reply:
> I'm fully OK to add it separately by myself (in this or separate testcase). I'm just afraid that it'll also just end with similar discussion.

Right, it might end up in a similar discussion and there is nothing wrong about. I am just against adding anything into samples that has not been verified or documented and this is just a basic build sanity check and nothing else, if everyone comes in and adds their favorite platforms, this will just get out of control and we will spend our time building samples. So, please submit a seperate PR adding nsim with a justification why it should be added and possible documentation of how it can be used with the tool, but blocking PRs becasue you want to add a platform is not going to get us anywhere.